src/erlcloud_aws.erl

```diff
@@ -796,14 +797,59 @@ service_host( <<"s3">>, Region ) ->
 service_host( <<"iam">>, <<"cn-north-1">> ) -> "iam.amazonaws.com.cn";
 service_host( <<"iam">>, <<"cn-northwest-1">> ) -> "iam.amazonaws.com.cn";
 service_host( <<"sdb">>, <<"us-east-1">> ) -> "sdb.amazonaws.com";
-service_host( <<"states">>, Region ) ->
-    binary_to_list( <<"states.", Region/binary, ".amazonaws.com">> );
 service_host( Service, <<"cn-north-1">> = Region ) when is_binary(Service) ->
     binary_to_list( <<Service/binary, $., Region/binary, ".amazonaws.com.cn">> );
-service_host( Service, <<"cn-northwest-1">> =Region ) when is_binary(Service) ->
+service_host( Service, <<"cn-northwest-1">> = Region ) when is_binary(Service) ->
     binary_to_list( <<Service/binary, $., Region/binary, ".amazonaws.com.cn">> );
-service_host( Service, Region ) when is_binary(Service) ->
-    binary_to_list( <<Service/binary, $., Region/binary, ".amazonaws.com">> ).
+% services which can have VPCe configured to mitigate cross-AZ traffic.
+% It's application level decision to use VPCe and configure those
+% magic can be done vi EC2 DescribeVpcEndpoints/filter by VPC/filter by AZ,
+% however permissions and describe* API throttling is not what we want to deal with here.
+service_host( Service, Region ) when is_binary(Service) andalso is_binary(Region) ->
+    Default = binary_to_list( <<Service/binary, $., Region/binary, ".amazonaws.com">> ),
+    get_host_vpc_endpoint(Service, Default).
+
+-spec get_host_vpc_endpoint(binary(), binary()) -> binary().
+%% take the list of possibly configured endpoints
+%% and pick the one which suites from our AZ.
+get_host_vpc_endpoint(Service, Default) when is_binary(Service) ->
+    ConfiguredEndpoints = proplists:get_value(Service,
```

Comment:
Split this into 2 calls, IMO

```erlang
VPCEndpointsByService = application:get_env(erlcloud, services_vpc_endpoints, []),
ConfiguredEndpoints = proplists:get_value(Service, ConfiguredEndpoints, []),
```

As you have it, it looks like it can return `undefined` and reach a case clause?